sql/database.go

```diff
@@ -292,16 +328,201 @@ func Open(uri string, opts ...Opt) (*sqliteDatabase, error) {
 	return db, nil
 }
 
+func ensureDBSchemaUpToDate(logger *zap.Logger, db *sqliteDatabase, config *conf) (*sqliteDatabase, error) {
+	before, after, err := config.schema.CheckDBVersion(logger, db)
+	switch {
+	case err != nil:
+		return db, fmt.Errorf("check db version: %w", err)
+	case before == after:
+		return db, nil
+	case before > after:
+		// TODO: this should be logged by the caller
+		logger.Error("database version is newer than expected - downgrade is not supported",
+			zap.Int("current version", before),
+			zap.Int("target version", after),
+		)
+		return db, fmt.Errorf("%w: %d > %d", ErrTooNew, before, after)
```

Comment:
I feel like this should be fixed before merging just removing the long from here is probably enough:

```suggestion
		return db, fmt.Errorf("%w: %d > %d", ErrTooNew, before, after)
```

The error bubbles up all the way to `sql.Open` and I believe every caller of that function will log and exit if it returns an error.

Reply:
Fixed